[#5183] nightly task to check geoserver registrations

devincowan opened the pull request:
Resolves #5183 
<!--

Please read, and add your text at the bottom of this message.

Thanks for contributing code to HydroShare. In order to maintain code quality and expedite this process, please assist the development team by making sure the following is present in this pull request.

For more information, see https://docs.google.com/document/d/1dzxqlZW5fKNEyQSeKiSFq-SmS-VOPCva95XXkBjPExs

-->

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

<!-- Enter steps that a QA engineer, stakeholder, or user documentation writer would follow to test the positive or "successful" case of the functionality your code provides or fixes -->

- [ ] Automated Testing

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Code coverage must not decrease so new functionality or code paths added during a bug fix must have appropriate tests written. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Sufficient User and Developer Documentation

<!-- Please email your positive test case lbrazil@cuahsi.org, who will make the decision regarding user documentation. -->

- [ ] Passing Jenkins Build

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Peer Code review and approval

<!-- This is the process by which a peer developer on the HydroShare team will read the changeset, provide feedback, and ultimately give a formal approval to the code before it passes PR status. -->

### Positive Test Case
1. [Enter positive test case here]

sblack-usu commented on hs_core/tasks.py:
```diff
@@ -374,6 +382,44 @@ def send_over_quota_emails():
+                res_url = current_site_url() + res.get_absolute_url()
+                failed_resources[res_url] = response
+    except Exception as e:
+        err_msg = f"Exception while updating web services. Error: {str(e)}\n"
```
This will terminate the process for any remaining resources in the list.  Would it be beneficial to try/except the update_web_service function call instead?

devincowan replied:
Good call thanks, fixed that

sblack-usu commented on hs_core/tasks.py:
```diff
@@ -374,6 +382,44 @@ def send_over_quota_emails():
+    # Check to ensure resources have updated web services registrations
+    if not resources:
+        cuttoff_time = timezone.now() - timedelta(days=1)
+        resources = BaseResource.objects.filter(updated__gte=cuttoff_time, raccess__public=True)
```
Not all public resources use geoserver.  Are resources that do not contain geoserver files creating noise in the logging for failed_resources below?

devincowan replied:
the web service middleware will return a successful response for resources that don't contain geospatial files

sblack-usu commented on hs_core/tasks.py:
```diff
@@ -374,6 +382,44 @@ def send_over_quota_emails():
+    err_msg = ""
+    try:
+        for res in resources:
+            response = update_web_services(
```
The only other feedback I have is that these network calls happen sequentially.  It is only really a concern for large batches of resources.  It may not be much of a concern because of geoserver being in the same network as well.

devincowan replied:
thanks @sblack-usu. I took the approach of distributing these network calls over 2 hours in https://github.com/hydroshare/hydroshare/pull/5186/commits/931d8ee39ea505e6947eb1651cb89d8a4b28895a

devincowan commented:
Thanks for the thoughtful suggestions @sblack-usu! I believe I addressed all of them.

devincowan commented:
@sblack-usu I realized that the way I made web services requests async and distributed was going to cause issues, so I fixed that in https://github.com/hydroshare/hydroshare/pull/5186/commits/7ae5dd1d60c7870fa08da15bdcb63a21929b93ff. PR is ready for re-review next week.

I also changed the base of this PR from `master` to `develop` because it's Friday and we can release this as part of 2.9 instead of additional reviews today! :) Rather than releasing this as 2.8.2 hotfix, I modified the [data services notebook](https://github.com/hydroshare/hs_data_services/blob/main/register_resource_test.ipynb) and ran it to blindly re-register ALL resources as @martinseul suggested.
